### ISSUE TITLE
guild.invites.fetch gives an unexpected result

### ISSUE THREAD

**Issue by STR1KE69** [discordjs/discord.js#6098]
**Please describe the problem you are having in as much detail as possible:**
         the code explains itself
**Include a reproducible code sample here, if possible:**

```js
const Discord = require('discord.js');
const client = new Discord.Client({ intents: ['GUILDS', 'GUILD_MEMBERS', 'GUILD_INVITES', 'GUILD_MESSAGES', 'DIRECT_MESSAGES'] });

const config = require("../database/config.json");
client.login(config.token);

var invites = {};

client.on("ready", () => {
  client.guilds.cache.get("GUILD_ID").invites.fetch().then(guildInvites => { // Fetch invitations from server...
    invites = guildInvites; // Refresh the list.
  });
});

client.on("guildMemberAdd", async (member) => { // Detect when a member arrives.

  var invite; // For later.
  member.guild.invites.fetch().then(guildInvites => { // Watch the invitations from the server...
    guildInvites.forEach((code) => { // For each server invite code...

      var invitation_before = invites.get(code.code); // Take "invitation_before" as being the list of invitations BEFORE the arrival of the member.
      var invitation_after = guildInvites.get(code.code); // Take "invitation_after" as the list of invitations AFTER the arrival of the member.

      if (invitation_before.uses !== invitation_after.uses) { // Compare if the selected code has the same number of uses or not...
        invite = invitation_after; // If so, this is the invitation used!
      };
    });

    console.log(invite) // expected: the invitation (object) with which the new member arrived. In reality: "undefined" => invitation_before.uses === invitation_after.uses.
  });
});
```

**Further details:**

- discord.js version: 13.0.0
- Node.js version: 14.17.0
- Operating system: Windows 10
- Priority this issue should have – please be realistic and elaborate if possible:

**Relevant client options:**

- partials: none
- gateway intents: 'GUILDS', 'GUILD_MEMBERS', 'GUILD_INVITES', 'GUILD_MESSAGES', 'DIRECT_MESSAGES']
- other: none

**Comment by BannerBomb** [comment]
It could be because the classes are mutable so since you're not cloning any of the invite classes from the cache that you fetch from the ready event they will change when the patch method is called. Fetching invites from the guild also looks in the cache before fetching from the API by default so since you aren't forcing the fetch you are just getting the same cached classes both times you fetch the invites.

Basically both invitation_before and invitation_after will contain the same data since the classes are mutable which causes `if (invitation_before.uses !== invitation_after.uses) {` to fail.

**Comment by STR1KE69** [comment]
> It could be because the classes are mutable so since you're not cloning any of the invite classes from the cache that you fetch from the ready event they will change when the patch method is called. Fetching invites from the guild also looks in the cache before fetching from the API by default so since you aren't forcing the fetch you are just getting the same cached classes both times you fetch the invites.
> 
> Basically both invitation_before and invitation_after will contain the same data since the classes are mutable which causes `if (invitation_before.uses !== invitation_after.uses) {` to fail.
how do i fix

**Comment by iCrawl** [comment]
__The issue tracker is only for bug reports and enhancement suggestions. If you have a question, please ask it in the [Discord server](https://discord.gg/bRCvFy9) instead of opening an issue – you will get redirected there anyway.__

**Comment by STR1KE69** [comment]
> **The issue tracker is only for bug reports and enhancement suggestions. If you have a question, please ask it in the [Discord server](https://discord.gg/bRCvFy9) instead of opening an issue – you will get redirected there anyway.**

I already asked but no one knew
